runtime/parser/declaration.go

```diff
@@ -861,7 +861,8 @@ func parseEventDeclaration(
 	// Skip the identifier
 	p.next()
 
-	parameterList, err := parseParameterList(p)
+	// if this is a `ResourceDestroyed` event (i.e., a default event declaration), parse default arguments
+	parameterList, err := parseParameterList(p, identifier.Identifier == ast.ResourceDestructionDefaultEventName)
```

Comment:
Maybe factor the expression out and give it a descriptive name to make the code a bit more self-explaining

```suggestion
	parseDefaultArguments := identifier.Identifier == ast.ResourceDestructionDefaultEventName
	parameterList, err := parseParameterList(p, parseDefaultArguments)
```